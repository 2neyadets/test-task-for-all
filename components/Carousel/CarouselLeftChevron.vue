<template lang="pug">
  .absolute.flex.flex-center.carousel__left-chevron
    Btn(
      @click.native="$emit('changeSlide', currentSlide - 1)"
      icon="chevron_left"
      :disable="currentSlide === 0"
    )
</template>

<script>

import Btn from '../Reusable/Btn'
export default {
  name: 'CarouselLeftChevron',
  components: { Btn },
  props: {
    currentSlide: {
      type: Number,
      required: true,
    },
    amountOfSlides: {
      type: Number,
      required: true,
    },
  },
  data () {
    return {
    }
  },
  computed: {},
  watch: {},
  created () {},
  mounted () {},
  updated () {},
  methods: {},
}
</script>

<style scoped lang="stylus">
  .carousel__left-chevron
    top 16px
    bottom 16px
    left 12px
</style>
