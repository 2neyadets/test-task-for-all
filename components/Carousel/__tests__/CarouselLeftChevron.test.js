import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import CarouselLeftChevron from '../CarouselLeftChevron'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('CarouselLeftChevron component', () => {
  let wrapper
  const currentSlide = 1,
    amountOfSlides = 3

  beforeEach(() => {
    wrapper = shallowMount(CarouselLeftChevron, {
      localVue,
      propsData: {
        currentSlide,
        amountOfSlides,
      },
    })
  })

  it('mounts without errors', done => {
    localVue.config.errorHandler = done // для отлавливания ошибок внутри nextTick, если нужно что-то выпонять внутри него
    expect.assertions(1)
    expect(wrapper).toBeTruthy()
    done()
  })

  it('can emit "changeSlide" event', () => {
    expect.assertions(2)
    wrapper.find('MyBtn-stub').trigger('click')
    expect(wrapper.emitted('changeSlide').length).toBe(1)
    expect(wrapper.emitted('changeSlide')[0]).toEqual([currentSlide - 1])
  })

  it('can be disabled', async () => {
    expect.assertions(1)
    wrapper.setProps({
      currentSlide: 0,
    })
    await localVue.nextTick()
    expect(wrapper.find('MyBtn-stub').attributes().disable).toBe('true')
  })

  it('match snapshot', () => {
    expect.assertions(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
