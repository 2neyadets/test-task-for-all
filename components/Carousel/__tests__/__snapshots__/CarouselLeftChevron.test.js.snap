// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselLeftChevron component match snapshot 1`] = `
<div
  class="absolute flex flex-center carousel__left-chevron"
>
  <mybtn-stub
    icon="chevron_left"
    label=""
    size="20"
    type="button"
  />
</div>
`;
